src/cloudant/design_document.py

```diff
@@ -86,14 +90,15 @@ def update_view(self, view_name, map_func, reduce_func=None):
         if view is None:
             msg = "View {0} does not exist in this design doc".format(view_name)
             raise CloudantArgumentError(msg)
-        view.map = map_func
-        if reduce_func is not None:
-            view.reduce = reduce_func
+        view = View(self, view_name, map_func, reduce_func, **kwargs)
         self.views.__setitem__(view_name, view)
 
     def delete_view(self, view_name):
         """
         Removes a View from the locally cached DesignDocument View dictionary.
+        Do not use this method if you wish to
+        delete a query index view.  Use
+        :func:`~cloudant.database.CloudantDatabase.delete_index` instead.
```

Comment:
what happens if they try it any way? Execption? Maybe call that out in the doc. Otherwise this is just a "do not push this button" type of button